dandi/dandiapi.py

```diff
@@ -295,6 +296,16 @@ def create_dandiset(self, name: str, metadata: Dict[str, Any]) -> "RemoteDandise
             self, self.post("/dandisets/", json={"name": name, "metadata": metadata})
         )
 
+    def match_schema_version(self, schema_version: Optional[str] = None) -> None:
+        if schema_version is None:
+            schema_version = get_schema_version()
+        server_schema_version = self.get("/info/")["schema_version"]
+        if server_schema_version != schema_version:
+            raise SchemaVersionError(
+                f"Server requires schema version {server_schema_version};"
+                f" client only has {schema_version}"
```

Comment:
```suggestion
                f" client supports {schema_version}. Upgrade of the dandischema and/or dandi client might be needed."
```
to provide some guidance to the user on what to do... 
(may be here we should remind with etelemetry, but we do not even track dandischema with it ATM... so let's not bother for this one)

Reply:
Done.